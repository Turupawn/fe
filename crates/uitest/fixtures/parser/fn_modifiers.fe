pub pub struct Foo {}
impl Foo {
    pub pub unsafe unsafe fn f() {}
    unsafe pub fn g() {}
    unsafe unsafe pub unsafe pub pub unsafe fn h() {}
}

trait T {
    fn f(self)
}

impl T for Foo {
    pub unsafe fn f(self) {}
}