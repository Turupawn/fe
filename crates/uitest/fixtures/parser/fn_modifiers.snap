---
source: crates/uitest/tests/parser.rs
expression: diags
input_file: crates/uitest/fixtures/parser/fn_modifiers.fe
---
error[1-0001]: duplicate `pub` modifier
  ┌─ fn_modifiers.fe:1:5
  │
1 │ pub pub struct Foo {}
  │     ^^^ unexpected

error[1-0001]: duplicate `pub` modifier
  ┌─ fn_modifiers.fe:3:9
  │
3 │     pub pub unsafe unsafe fn f() {}
  │         ^^^ unexpected

error[1-0001]: duplicate `unsafe` modifier
  ┌─ fn_modifiers.fe:3:20
  │
3 │     pub pub unsafe unsafe fn f() {}
  │                    ^^^^^^ unexpected

error[1-0001]: `pub` modifier must come before `unsafe`
  ┌─ fn_modifiers.fe:4:12
  │
4 │     unsafe pub fn g() {}
  │            ^^^ unexpected

error[1-0001]: duplicate `unsafe` modifier
  ┌─ fn_modifiers.fe:5:12
  │
5 │     unsafe unsafe pub unsafe pub pub unsafe fn h() {}
  │            ^^^^^^ unexpected

error[1-0001]: `pub` modifier must come before `unsafe`
  ┌─ fn_modifiers.fe:5:19
  │
5 │     unsafe unsafe pub unsafe pub pub unsafe fn h() {}
  │                   ^^^ unexpected

error[1-0001]: duplicate `unsafe` modifier
  ┌─ fn_modifiers.fe:5:23
  │
5 │     unsafe unsafe pub unsafe pub pub unsafe fn h() {}
  │                       ^^^^^^ unexpected

error[1-0001]: duplicate `pub` modifier
  ┌─ fn_modifiers.fe:5:30
  │
5 │     unsafe unsafe pub unsafe pub pub unsafe fn h() {}
  │                              ^^^ unexpected

error[1-0001]: duplicate `pub` modifier
  ┌─ fn_modifiers.fe:5:34
  │
5 │     unsafe unsafe pub unsafe pub pub unsafe fn h() {}
  │                                  ^^^ unexpected

error[1-0001]: duplicate `unsafe` modifier
  ┌─ fn_modifiers.fe:5:38
  │
5 │     unsafe unsafe pub unsafe pub pub unsafe fn h() {}
  │                                      ^^^^^^ unexpected

error[1-0001]: `pub` modifier is not allowed in this block
   ┌─ fn_modifiers.fe:13:5
   │
13 │     pub unsafe fn f(self) {}
   │     ^^^ unexpected

error[1-0001]: `unsafe` modifier is not allowed in this block
   ┌─ fn_modifiers.fe:13:9
   │
13 │     pub unsafe fn f(self) {}
   │         ^^^^^^ unexpected


